minifi/minifi-bootstrap/src/main/java/org/apache/nifi/minifi/bootstrap/service/BootstrapFileProvider.java

```diff
@@ -0,0 +1,234 @@
+        } catch (Exception e) {
+            LOGGER.warn("Failed to set permissions so that only the owner can read status file {}; "
+                + "this may allows others to have access to the key needed to communicate with MiNiFi. "
+                + "Permissions should be changed so that only the owner can read this file", statusFile);
+        }
+
+        try (FileOutputStream fos = new FileOutputStream(statusFile)) {
+            minifiProps.store(fos, null);
+            fos.getFD().sync();
+        }
+
+        LOGGER.debug("Saving MiNiFi properties to {}", statusFile);
+        logProperties("Saved MiNiFi", minifiProps);
+    }
+
+    private void writePidFile(String pid) throws IOException {
+        File pidFile = getPidFile();
+        if (pidFile.exists() && !pidFile.delete()) {
+            LOGGER.warn("Failed to delete {}", pidFile);
+        }
+
+        if (!pidFile.createNewFile()) {
+            throw new IOException("Failed to create file " + pidFile);
+        }
+
+        try {
+            Set<PosixFilePermission> perms = new HashSet<>();
+            perms.add(PosixFilePermission.OWNER_READ);
+            perms.add(PosixFilePermission.OWNER_WRITE);
+            Files.setPosixFilePermissions(pidFile.toPath(), perms);
+        } catch (Exception e) {
+            LOGGER.warn("Failed to set permissions so that only the owner can read pid file {}; "
+                + "this may allows others to have access to the key needed to communicate with MiNiFi. "
+                + "Permissions should be changed so that only the owner can read this file", pidFile);
+        }
+
+        try (FileOutputStream fos = new FileOutputStream(pidFile)) {
+            fos.write(pid.getBytes(StandardCharsets.UTF_8));
+            fos.getFD().sync();
+        }
+
+        LOGGER.debug("Saved Pid {} to {}", pid, pidFile);
+    }
+
+    private File getBootstrapFile(String fileName) throws IOException {
+        File configFileDir = Optional.ofNullable(System.getProperty(MINIFI_PID_DIR_PROP))
+            .map(String::trim)
+            .map(File::new)
+            .orElseGet(() -> {
+                File confDir = bootstrapConfigFile.getParentFile();
+                File nifiHome = confDir.getParentFile();
+                return new File(nifiHome, DEFAULT_PID_DIR);
+            });
+
+        FileUtils.ensureDirectoryExistAndCanAccess(configFileDir);
+        File statusFile = new File(configFileDir, fileName);
+        LOGGER.debug("Run File: {}", statusFile);
+
+        return statusFile;
+    }
+
+    private void logProperties(String type, Properties props) {
+        if (LOGGER.isDebugEnabled()) {
+            LOGGER.debug("{} properties: {}", type, props.entrySet()
+                .stream()
+                .filter(e -> {
+                    String key = ((String) e.getKey()).toLowerCase();
+                    return !"secret.key".equals(key) && !key.contains("password") && !key.contains("passwd");
```

Comment:
This check is helpful, but it seems prone to missing new values. It seems better to avoid logging properties altogether, or incorporating something with a property name enum that indicates sensitive status.